src/olympia/reviewers/tests/test_views.py

```diff
@@ -6020,6 +6020,76 @@ def test_abuse_reports_resolved_as_approve_with_approve_latest_version_action(
         assert decision.action == DECISION_ACTIONS.AMO_APPROVE_VERSION
         self.assertCloseToNow(decision.action_date)
 
+    def test_force_enable_shows_versions_to_reenabled(self):
+        v1 = self.addon.current_version
+        v2 = version_factory(addon=self.addon)
+        v3 = version_factory(addon=self.addon)
+        v4 = version_factory(addon=self.addon, channel=amo.CHANNEL_UNLISTED)
+        self.addon.update(status=amo.STATUS_DISABLED)
+        v1.file.update(
+            status=amo.STATUS_DISABLED,
+            original_status=amo.STATUS_APPROVED,
+            # We don't want to re-enable a version the developer disabled
+            status_disabled_reason=File.STATUS_DISABLED_REASONS.DEVELOPER,
+        )
+        v2.file.update(
+            status=amo.STATUS_DISABLED,
+            original_status=amo.STATUS_APPROVED,
+            # we also don't want to re-enable a version we rejected
+            status_disabled_reason=File.STATUS_DISABLED_REASONS.NONE,
+        )
+        v3.file.update(
+            status=amo.STATUS_DISABLED,
+            original_status=amo.STATUS_APPROVED,
+            # but we do want to re-enable a version we only disabled with the addon
+            status_disabled_reason=File.STATUS_DISABLED_REASONS.ADDON_DISABLE,
+        )
+        v4.file.update(
+            status=amo.STATUS_DISABLED,
+            original_status=amo.STATUS_AWAITING_REVIEW,
+            # awaiting review versions could also be re-enabled
+            status_disabled_reason=File.STATUS_DISABLED_REASONS.ADDON_DISABLE,
+        )
+
+        response = self.client.get(self.url)
+        self.assertContains(response, '2 version(s) will be re-enabled:')
+        doc = pq(response.content)
+        items = doc('li.version-that-would-be-reenabled')
+        assert len(items) == 2
+        assert v3.version in items[1].text_content()
```

Comment:
These assertions are a bit tough for me to interpret. I think I understand, but it takes longer than it should. Here are some ideas on how this could be improved for readability:
- you could combine the split assertions for the version and the text via a regex assertion.
- can you name items[x] to indicate what exactly is the difference between item 0 and 1?
- it might be good to group the assertions based on the version you are asserting against rather than the type of content you're looking for. It might be easier to understand.

Reply:
my feelings on regex are summed up by the quote
> Some people, when confronted with a problem, think “I know, I’ll use regular expressions.” Now they have two problems.

🤣

Reply:
reworked the assertions somewhat; hope that helps